### ISSUE TITLE
Docker compose configuration

### ISSUE THREAD

**Issue by infogulch** [terminusdb/terminusdb-quickstart#5]
I prefer to manage infrastructure via declarative configuration as opposed to running a script, as it allows me to fully understand and track the desired state with source control and replicate a fully integrated deployment more reliably.

I would like to have a `docker-compose.yml` file that describes an example working terminusdb docker configuration.  Specifically, I would hope that this configuration be maintained here -- if not officially recommended, at least as a starting point.

I've tried to build this myself, but I ran into issues that prevented it from working.  I also tried the script and it works fine, but that just says to me that I don't understand what it takes to administer terminusdb and I don't have a grasp on what is important, which makes me even more leery of just relying on the script to manage everything.  No magic, please.

It's been a few weeks since I last approached the problem of defining a desired state configuration for terminusdb, but I would still like to tackle it if the team would be interested in such a contribution.  If we can get a docker compose-based configuration working, would the terminusdb team desire to keep it here, even unofficially?

I would provide more details about the specific problems I've encountered once I jump back into it.

TL;DR: I would like to docker-compose-ify terminusdb-quickstart; vibe check?

**Comment by dmytri** [comment]
I have no objection to a docker compose file.

The main purpose of this script is:

- To make it easy to run TerminusDB Server anywhere, including various Linux distros, MacOS, and windows, as a place holder for the "real distributions" that we will make in later stages of our development, like debs, rpms, flatpaks, exes, dmgs, etc

- To be very configurable, but also have sane defaults that "work out of the box."

- To allow developers working on TerminusDB's server and console components and SDKs, to run the server with various versions of its parts, primarily different versions the server and console components.

- To use in our CI/CD testing pipelines for full end-to-end testing.

I think it would be very cool to have a a great example docker compose file for running TerminusDB, but I think it makes sense for such an example compose file to be as simple as possible as a starting point, which does not need to fulfill the goals of this script above.

Of course, this doesn't mean we couldn't have such a file in this repo tho, so look forward to seeing what you have in mind.

Thanks so much for your thoughtful comment and, of course, for trying TerminusDB!!

**Comment by infogulch** [comment]
Thank you for your response! I especially appreciate that you outlined the motivation/purpose of the script as it exists today, your plans for how you hope to solve those problems as the project matures, and how my suggested addition could map onto that.

I'll update here when I have something more substantive.  Thank you!
